File: grulla99/binarySearch/1822.py
import sys

input = sys.stdin.readline

n, m = map(int, input().split())

a = list(map(int, input().split()))
b = list(map(int, input().split()))

a.sort()
b.sort()


def binary_sarch(a, b):
    result = []

    for i in a:
        start = 0
        end = len(b) - 1
        found = False
        while start <= end:
            mid = (start + end) // 2
            if b[mid] == i:
                found = True
                break
            elif b[mid] < i:
                start = mid + 1
            else:
                end = mid - 1
        if not found:
            result.append(i)
    return result

result = binary_sarch(a, b)

if len(result) == 0:
    print(0)
else:
    print(len(result))
    print(" ".join(map(str, result)))

File: 하은/이분탐색/1822.js
const solution = () => {
    const fs = require('fs');
    const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
    const input = fs.readFileSync(filePath, 'utf-8').split('\n');

    const [N_A, N_B] = input[0].split(' ').map(Number);

    const A = input[1]
        .split(' ')
        .map(Number)
        .sort((a, b) => a - b);

    const B = input[2]
        .split(' ')
        .map(Number)
        .sort((a, b) => a - b);

    function binarySearch(target) {
        let start = 0;
        let end = N_B - 1;

        while (start <= end) {
            let mid = Math.floor((start + end) / 2);

            if (target === B[mid]) return;
            else if (target > B[mid]) start = mid + 1;
            else end = mid - 1;
        }
        return target;
    }

    const res = A.map((v) => binarySearch(Number(v))).filter((a) => a);

    return res.length ? `${res.length}\n`.concat(res.join(' ')) : '0';
};

console.log(solution());
